fuzz-test-suite/dateparserfuzzer.cpp

```diff
@@ -0,0 +1,24 @@
+#include <ql/time/asx.hpp>
+#include <ql/time/date.hpp>
+#include <ql/time/ecb.hpp>
+#include <ql/time/imm.hpp>
+#include <ql/time/timeunit.hpp>
+#include <ql/utilities/dataparsers.hpp>
+#include <string>
+
+#ifndef __clang__
+#pragma message("Fuzzer headers are available from clang, other compilers have not been tested.")
+#endif
+#include <fuzzer/FuzzedDataProvider.h>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+    FuzzedDataProvider fdp(data, size);
+    constexpr int kMaxString = 100;
+    try {
+        (void)QuantLib::DateParser::parseFormatted(fdp.ConsumeRandomLengthString(kMaxString), "%Y-%m-%d");
```

Comment:
So it turns out there is a bug in parseFormatted, which results in an integer overflow, specifically if the string input is empty i.e. `""` you'll end up with an integer overflow when parsing. I compiled the fuzzer with `-fsanitize=undefined -fno-sanitize-recover=all`, and it quickly found the bug.

Reply:
So this simplest way to reproduce this would be something like;
```c++
#include <ql/time/asx.hpp>
#include <ql/time/date.hpp>
#include <ql/time/ecb.hpp>
#include <ql/time/imm.hpp>
#include <ql/time/timeunit.hpp>
#include <ql/utilities/dataparsers.hpp>
#include <string>

int main() {
  (void)QuantLib::DateParser::parseFormatted("", "%Y-%m-%d");
  return 0;
}
```
Then compile with the flags as above `-fsanitize=undefined -fno-sanitize-recover=all`. Not a security risk, but it might give you some funky results.